examples/panel/book-recommender/app.py

```diff
@@ -0,0 +1,132 @@
+"""
+Demonstrates how to use the `ChatInterface` and a `callback` function to respond.
+
+The chatbot Assistant echoes back the message entered by the User.
+
+https://github.com/holoviz-topics/panel-chat-examples/blob/main/docs/examples/basics/basic_chat.py
+"""
+
+import panel as pn
+from openai import OpenAI
+from scipy.spatial import KDTree
+import numpy as np
+
+from rag_book_recommender import get_book_description_by_title, EmbeddingsStore, get_authors, get_embeddings
+
+
+client = OpenAI()
+
+pn.extension()
+
+store = EmbeddingsStore()
+all_authors = get_authors()
+
+
+def detect_author(user_query):
+    system_prompt = f"""
+    You're a system that determines the author in user query. 
+    
+    You need to return only he author name.Please fix any typo if possible
+"""
+    response = client.chat.completions.create(
+        model="gpt-3.5-turbo",
+        messages=[
+            {"role": "system", "content": system_prompt},
+            {"role": "user", "content": "What are some books by Sandra Boynton"},
+            {"role": "system", "content": "Sandra Boynton"},
+            {"role": "user", "content": user_query},
+        ],
+        seed=42,
+        n=1,
+    )
+    author = response.choices[0].message.content.upper()
+    return author if author in all_authors else ""
+
+
+def book_recommender_agent(user_query, verbose=False):
+    """An agent that can retrieve news by topic and summarizes them"""
```

Comment:
this docstring is innacurate

Reply:
[fixed](https://github.com/ploomber/doc/pull/127/files#diff-99d50d468b15d576674758b1fbde220b390f380dc2398dd8b4024a04218dd2ccR86-R87)